### DISCUSSION TITLE
Stuck at mamba installation

### DISCUSSION THREAD

**Discussion by debekkerlab** [biostars/biostar-handbook-issues#219] [Q&A]
**Both in the quick install and the step by step install I seem to bump into the same problem: the mamba installation seems to get stuck at "solving the environment". For both installs I noticed the following prompts:**

$ conda install mamba -n base -c conda-forge -y

Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: / 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
Examining conflict for urllib3 requests cryptography pyopenssl: : 42it [02:27,  Examining conflict for colorama conda-package-handling tqdm: : 42it [02:28,  1.1Examining conflict for colorama conda-package-handling tqdm: : 43it [02:28,  1.1Examining conflict for conda requests: : 43it [02:29,  1.14s/it]                Examining conflict for conda-package-handling conda tqdm: : 44it [02:29,  1.13itExamining conflict for conda-package-handling conda tqdm: : 45it [02:29,  1.26itExamining conflict for conda-package-handling conda: : 45it [02:29,  1.26it/sfailed                                                                           

UnsatisfiableError: The following specifications were found to be incompatible with a past
explicit spec that is not an explicit spec in this operation (conda):

  - mamba -> conda[version='4.6.*,<4.13.0|4.6.*|4.7.*,<4.13.0|>=4.7.12,<4.13.0|>=4.8,<4.13.0|>=4.7.12,<4.8']

The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

**After, indeed several minutes it spits out a ton of conflicts for seemingly everything, examples:**

Package packaging conflicts for:
pip -> wheel -> packaging[version='>=20.2']
wheel -> packaging[version='>=20.2']

Package urllib3 conflicts for:
urllib3
conda[version='>=4.13.0'] -> requests[version='>=2.18.4,<3'] -> urllib3[version='>=1.21.1,<1.22|>=1.21.1,<1.23|>=1.21.1,<1.24|>=1.21.1,<1.25|>=1.21.1,<1.26,!=1.25.0,!=1.25.1|>=1.21.1,<1.27']
requests -> urllib3[version='>=1.21.1,<1.22|>=1.21.1,<1.23|>=1.21.1,<1.24|>=1.21.1,<1.25|>=1.21.1,<1.26,!=1.25.0,!=1.25.1|>=1.21.1,<1.27']

Package ipaddress conflicts for:
cryptography -> ipaddress
requests -> urllib3[version='>=1.21.1,<1.27'] -> ipaddress
pyopenssl -> cryptography[version='>=2.8'] -> ipaddress
urllib3 -> ipaddress

**and so on. Total beginner here so I am kinda stuck now without knowing what to do. Anyone have any suggestions what I am doing wrong here?**

**Comment by ialbert** [comment]
what is the operating system that you are using?

mamba is one of the most basic tools, it is not even bioinformatics related - but evidently not being to be able to install it is a big deal

Mamba is solely used to make installations faster:

https://github.com/mamba-org/mamba

never thought that mamba itself would cause problems.

**Reply by debekkerlab** [reply]
Thanks so much for the quick response. I’m using the latest Mac OS 12.4. On a “regular” MacBook Pro. No fancy M1 chip yet.

**Comment by ialbert** [comment]
can you double  check that you have updated XCode

some of those instructions in the MacOS setup:

https://www.biostarhandbook.com/setup-macos.html#setup-macos

when MacOS gets updated to latest new distribution, not all packages get the latest version, and occasionally issues arise

tomorrow I will try to reinstall on a brand new MacOS myself

**Reply by debekkerlab** [reply]
I installed the latest XCode after I updated my OS so you’d think it would be okay? But I’ll check the link you posted as well. Thanks again for trying to help and solve this!
